### PR TITLE
Update Gravity schema to include ViewingRooms [GALL-2712]

### PR DESCRIPTION
**Pull request by mdole**
Following up on Gravity work to create ViewingRoom types + subtypes (see https://github.com/artsy/gravity/pull/12904, https://github.com/artsy/gravity/pull/12915), this PR copies Gravity's schema over MP's copy of `gravity.graphql`.

We also discovered that `stitching/gravity/schema.ts` automatically renames root fields to have `_unused_gravity...` in front of them. Our workaround was to add an explicit `allowList` of fields - @zephraph we'd love to get your input on this as we're not sure if it's best practice. Thanks!

[Jira ticket](https://artsyproduct.atlassian.net/browse/GALL-2712)

### CODE REVIEW

File: src/lib/stitching/gravity/schema.ts [hunk 1]
```diff
@@ -50,7 +52,7 @@ export const executableGravitySchema = () => {
     // We have the same restrictions for root, so let's prefix
     // for now
     new RenameRootFields((type, name, _field) => {
-      if (type === "Query") {
+      if (type === "Query" && !allowList.includes(name)) {
```
Comment (iskounen) [inline src/lib/stitching/gravity/schema.ts:55]: @zephraph this is the change that we'd like to get some advice on. We added a new `viewingRoom` type to the gravity schema, and tried to query for it in Metaphysics' graphiql. What we found instead in the Docs tab was an `_unused_gravity_viewingRoom` type.

We were hoping that you may be able to shine some light on the origins of `_unused_gravity_` being prepended to types in the gravity schema, so that we can determine if this change makes sense.

Comment (zephraph) [reply src/lib/stitching/gravity/schema.ts:55]: I’m trying to dig up more context. I believe this was to a aid in cleanup of unused fields by giving something for a transformer to hook off of. 

I’ll get back with you once I have some context.

Comment (mdole) [reply src/lib/stitching/gravity/schema.ts:55]: thanks! much appreciated 🙏

Comment (zephraph) [reply src/lib/stitching/gravity/schema.ts:55]: Okay, while I haven't found the exact situation in which it was added, it was done as a part of https://github.com/artsy/metaphysics/pull/1854.

One of the things @alloy was trying to do generally is to clean up the root fields that were in the schema. _Especially_ root fields that weren't being used. Gravity actually had a lot of those. There was a transformer that went through and removed all the `_unused_gravity_*` fields from the final schema.

File: _schema.graphql [hunk 2]
```diff
@@ -5277,6 +5277,31 @@ type DaySchedule {
...
+
+# Autogenerated return type of DebitCommissionExemption
+type DebitCommissionExemptionPayload {
+  # The amount of GMV in sale currency to not charge commission on in this transaction
+  amountOfExemptGmvOrError: MoneyOrErrorUnion!
+
+  # A unique identifier for the client performing the mutation.
+  clientMutationId: String
+}
+
```
Comment (zephraph) [inline _schema.graphql:5304]: There's a _lot_ more in the schema now that I wouldn't necessarily expect to see. Was adding all this intentional or just a consequence of copying over the gravity schema?

File: src/data/gravity.graphql [hunk 3]
```diff
@@ -412,6 +511,11 @@ type Query {
   """
   artworks(ids: [ID!]!): [Artwork!]
 
+  """
+  Find an identity verification by id
+  """
+  identityVerification(id: ID!): IdentityVerification
+
```
Comment (zephraph) [inline src/data/gravity.graphql:518]: cc @dleve123 did we ever intend for identity verifications to be exposed via MP's schema?

File: _schema.graphql (hunk 2 above)
Comment (zephraph) [reply _schema.graphql:5304]: This is the V1 schema. 😬. Ignore.

File: src/data/gravity.graphql (hunk 3 above)
Comment (zephraph) [reply src/data/gravity.graphql:518]: This is the v1 schema, might not be relevant.

**Comment by zephraph** [comment]
Sorry, I dropped some concerns about schema additions, but those were mostly to the v1 schema. Wasn't paying close enough attention to the file I was commenting on.

File: src/lib/stitching/gravity/schema.ts (hunk 1 above)
Comment (alloy) [reply src/lib/stitching/gravity/schema.ts:55]: Correct; iirc you could probably just drop all of them when they are stitched in, as long as at least 1 type remains–which it sounds like you now have (?).

File: src/lib/stitching/gravity/schema.ts [hunk 4]
```diff
@@ -8,6 +8,8 @@ import {
 } from "graphql-tools"
 import { readFileSync } from "fs"
 
+const allowList = ["viewingRoom"]
```
Comment (damassi) [inline src/lib/stitching/gravity/schema.ts:11]: Why is this needed? Are fields stitched via Grav by default hidden?

File: _schema.graphql (hunk 2 above)
Comment (mdole) [reply _schema.graphql:5304]: @zephraph I think these changes are in the v2 schema as well though - this was all work that was done in Gravity back in Jan/Feb that we never added to MP (we decided at the time to hit Gravity's GQL endpoint directly from Exchange), so my guess is that MP's Gravity schema just hasn't been updated since then.

File: _schema.graphql [hunk 5]
```diff
@@ -8076,7 +8082,7 @@ input MoneyInput {
 }
 
 # A money object with amount in minor currency and a currency code, or an error object
-union MoneyOrErrorUnion = Error | Money
+union MoneyOrErrorUnion = Error
```
Comment (alloy) [inline _schema.graphql:8085]: Everything’s free! 💸

File: src/lib/stitching/gravity/schema.ts (hunk 1 above)
Comment (iskounen) [reply src/lib/stitching/gravity/schema.ts:56]: Thanks for digging into that. Why is it problematic for those unused root fields to be in the schema?

File: src/data/gravity.graphql (hunk 3 above)
Comment (dleve123) [reply src/data/gravity.graphql:518]: This is for the Gravity component of the composite schema – so not sure about the v1 part of your comment above?

Regardless, @mdole and I are ignoring this type from Gravity as it's legacy/usused: https://github.com/artsy/metaphysics/pull/2280/commits/00d367e1c30e60700625000aa2e052933a76005d. We tested out a graphql-all-the-way-down approach, but ended up fronting identity verifications from Gravity via REST instead.

The auctions team will own removing those types from Gravity, but that's decoupled from this PR now.

File: src/lib/stitching/gravity/schema.ts (hunk 1 above)
Comment (zephraph) [reply src/lib/stitching/gravity/schema.ts:57]: Our [schema design](https://github.com/artsy/README/blob/0666a71997b5210315f70a34971f52c786ff8f53/playbooks/graphql-schema-design.md#root-fields) document talks about this a bit. 

Root fields are the entry points into our data and we want to reduce the overall number of entrypoints to high level domain objects. This is mostly to help keep the overall complexity of our schema down. We definitely don't want to add things that aren't being used. When you add something someone is likely to use it down the road. When you decide that the thing you added isn't right for the original problem, then you either have to add something new or change the original one... and depending on who is using it, that can get problematic. 

To boil it all down, schema designs can be hard to change later. Let's only build out what we need and only when we need it.

File: _schemaV2.graphql [hunk 6]
```diff
@@ -3601,6 +3601,31 @@ type DaySchedule {
   dayOfWeek: String
 }
 
+# Autogenerated input type of DebitCommissionExemption
+input DebitCommissionExemptionInput {
```
Comment (mzikherman) [inline _schemaV2.graphql:3605]: What are all these changes? Are these things added to Gravity that weren't stitched into MP, and now are? If so, we might leave them out.

File: _schema.graphql [hunk 7]
```diff
@@ -8040,14 +8065,18 @@ type Money {
   major: Float!
 }
 
+# A simple type to hold an amount and currency code
 input MoneyInput {
-  # amount in dollars or major unit
-  amount: Float!
+  # Amount in the minor unit of currency
+  amountMinor: Int!
```
Comment (iskounen) [inline _schema.graphql:8071]: This schema doesn't match the[ v1 field definition of `MoneyInput`](https://github.com/artsy/metaphysics/blob/f885199a5ababedcd0b81fdbd55395dc48f9a98c/src/schema/v1/fields/money.ts#L146), and that's causing some v1 tests to fail for BNMO mutations.

Depending on why this change was made, I think that we will either need to fix the schema or the field definition to make them match.

File: _schemaV2.graphql [hunk 8]
```diff
@@ -5441,6 +5466,18 @@ type Money {
   ): Float!
 }
 
+# A simple type to hold an amount and currency code
+input MoneyInput {
+  # Amount in the minor unit of currency
+  amountMinor: Int!
```
Comment (iskounen) [inline _schemaV2.graphql:5472]: This schema doesn't match the [v2 field definition of `MoneyInput`](https://github.com/artsy/metaphysics/blob/f885199a5ababedcd0b81fdbd55395dc48f9a98c/src/schema/v2/fields/money.ts#L170), which oddly isn't causing any v2 tests to fail, but should be looked at.

File: src/lib/stitching/gravity/schema.ts (hunk 1 above)
Comment (iskounen) [reply src/lib/stitching/gravity/schema.ts:57]: I love the idea of curating the high level domain objects that we expose as entry-points. That makes a lot of sense to me.

File: _schemaV2.graphql (hunk 6 above)
Comment (iskounen) [reply _schemaV2.graphql]: @mzikherman we decided to prevent these unused things from being stitched in, and did so by splitting `removeTypes` into:

1. `duplicateTypes`: a list of types that are filtered because they override Metaphysics, and
2. `unusedTypes`: a list of types (and inputs) that are filtered because they are not used by Metaphysics

File: _schema.graphql (hunk 5 above)
Comment (alloy) [reply _schema.graphql:8052]: Perhaps there was an internal discussion about this, but I don’t quite understand this change, it should be a union of an error and a money object, right?

Comment (mdole) [reply _schema.graphql:8052]: @alloy fair question! Yes, this is a `Error | Money` union in Gravity's GQL types/schema. In this case, the `Money` portion was removed automatically during schema stitching after we specifically stopped overwriting MP's `Money` type with Gravity's `Money` type: https://github.com/artsy/metaphysics/blob/83d8dc976142acaaffac9b65a550bee057a0cff2/src/lib/stitching/gravity/schema.ts#L24-L33

We couldn't quickly figure out how to remove the `MoneyOrErrorUnion` completely so punted on it for now - didn't seem like it was worth slowing down this PR since we don't expect `MoneyOrErrorUnion` to actually be used in MP. But we're also pretty new to MP! Open to feedback
